src/graphics/text.c

```diff
@@ -273,53 +275,58 @@ static void number_to_string(uint8_t *str, int value, char prefix, const char *p
         postfix++;
     }
     str[offset] = 0;
+    return offset;
 }
 
 int text_draw_number(int value, char prefix, const char *postfix, int x_offset, int y_offset, font_t font)
 {
-    uint8_t str[100];
+    uint8_t str[NUMBER_BUFFER_LENGTH];
     number_to_string(str, value, prefix, postfix);
     return text_draw(str, x_offset, y_offset, font, 0);
 }
 
 int text_draw_number_colored(int value, char prefix, const char *postfix, int x_offset, int y_offset, font_t font, color_t color)
 {
-    uint8_t str[100];
+    uint8_t str[NUMBER_BUFFER_LENGTH];
     number_to_string(str, value, prefix, postfix);
     return text_draw(str, x_offset, y_offset, font, color);
 }
 
 int text_draw_money(int value, int x_offset, int y_offset, font_t font)
 {
-    uint8_t str[100];
-    number_to_string(str, value, '@', " Dn");
+    uint8_t str[NUMBER_BUFFER_LENGTH];
+    int money_len = number_to_string(str, value, '@', " ");
+    const uint8_t *postfix = lang_get_string(6, 0);
+    if (postfix && money_len < NUMBER_BUFFER_LENGTH - 1) {
```

Comment:
You don't really need the second part of the `if`.

`string_copy` will truncate the string if it is too long, which is better since at least is shows some text.

So, my suggestion is to just do the following:

```suggestion
    if (postfix) {
```

Reply:
I added it, becuase string_copy() doesn't work very robust when maxlength == 0, in that case it tries to rewrite symbol before the 'src' pointer
So this condition checks that there is at least 1 free symbol.
I do understand that there will be corect place in src[-1], but I felt like not relying on it

Reply:
I get what you mean, but the truth is `string_copy` already makes a few assumptions to save CPU cycles (like `src` and `dst` never being null pointers) which in the current type-safe world may send some shivers down the spine. In fact, if you mess too much with the translation files, you *will* crash the game.

However, since we more or less control the strings that get fed into `string_copy`, it should be safe to use.

Reply:
Alright, if we take such assumptions, than we can get rid of that check